src/Core/src/Core/TrimmerHelper.cs

```diff
@@ -0,0 +1,20 @@
+﻿using System;
+using System.Diagnostics.CodeAnalysis;
+
+namespace Microsoft.Maui
+{
+	class TrimmerHelper
+	{
+		public const string HotReload = "Hot Reload is not supported in Release (trimmed) scenarios.";
+		public const string ConcurrentDictionary = "We cannot specify DynamicallyAccessedMemberTypes on ConcurrentDictionary.GetOrAdd().";
+		public const string MakeGenericType = "We will need to suppress this warning until we can remove usage of Type.MakeGenericType().";
+
+		/// <summary>
+		/// Calls Type.GetType(), but with DynamicallyAccessedMemberTypes.Interfaces|PublicParameterlessConstructor.
+		/// This allows illink to statically analyze System.Reflection usage.
+		/// </summary>
+		[UnconditionalSuppressMessage("Trimming", "IL2073", Justification = "We cannot specify DynamicallyAccessedMemberTypes on object.GetType().")]
+		[return: DynamicallyAccessedMembers(DynamicallyAccessedMemberTypes.Interfaces | DynamicallyAccessedMemberTypes.PublicParameterlessConstructor)]
+		public static Type GetType(object obj) => obj.GetType();
```

Comment:
This doesn't fix anything - it's basically just a way to suppress the warning with no effect on trimming.
The annotation on the return value has no effect on trimming - it will simply make subsequent usages of the value not produce more warnings.

Basically what this does is that it replaces suppressions on all the callsites to this method with one suppression here.

Reply:
Yes, I was unsure what to do here.

If you have a method `foo(object o)` that needs `DynamicallyAccessedMemberTypes.PublicParameterlessConstructor`, for example. Let's say it just calls something like `Activator.CreateInstance(o.GetType())`.

How do you solve the warning? `DynamicallyAccessedMembers` only works for for `string` or `Type`? Maybe there isn't actually a way to handle this case?

Reply:
There's no solution for `foo(object o)` which in turn calls `o.GetType()` - there's no way static analysis tools can figure out which type is that going to be (at least reasonably).

If the type of the object is at least partially known statically, this is potentially doable. For example if it's `foo(IMyInterface o)` and then `Activator.CreateInstance(o.GetType())`, it's possible to add the `DynamicallyAccessedMembers` attribute onto the `IMyInterface` type. The meaning of this is that all types which implement such interface must fulfill the requirements from the attribute - the analysis should then figure out the `GetType` call. There are potential downsides to this solution though:
* Depending on the annotation and the interface/basetype this can include LOT of code in the app. So ideally this should be applied to a relatively specific interface/basetype
* It will only work if the type which implements the interface is actually used somewhere else in the app - this should be the case already, or you should get another warning (like some types of DI would go and call `Assembly.GetTypes`, this would break this, but it will also produce a separate warning).
* You must "own" the interface/basetype - in order to add the attribute to it.